crates/ty_server/src/session/settings.rs

```diff
@@ -7,6 +7,24 @@ use serde::Deserialize;
 /// Maps a workspace URI to its associated client settings. Used during server initialization.
 pub(crate) type WorkspaceSettingsMap = FxHashMap<Url, ClientSettings>;
 
+// TODO(dhruvmanila): We need to mirror the "python.*" namespace on the server side but ideally it
+// would be useful to instead use `workspace/didChangeConfiguration` notification instead. This
+// would be then used to get all settings and not just the ones in "python.*".
+
+#[derive(Debug, Deserialize, Default)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+#[serde(rename_all = "camelCase")]
+struct Python {
+    ty: Option<Ty>,
+}
+
+#[derive(Debug, Deserialize, Default)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+#[serde(rename_all = "camelCase")]
+struct Ty {
+    disable_language_services: Option<bool>,
+}
+
```

Comment:
An alternative here is to avoid introducing "Python" struct and directly use `disableLanguageServices` in `ClientSettings` like:

```rs
struct ClientSettings {
	disable_language_services: Option<bool>
}
```

The disadvantage for this is that in VS Code the setting would be `python.ty.disableLanguageServices` while in other editors it would just be `disableLanguageServices`.

This PR keeps it consistent across editors and this is what we want as well because once `workspace/didChangeConfiguration` notification support is added, we'd directly query the client (editor) to get this setting value instead of mirroring the structure here.